File: Android Basic/Android-First-Game-Java-Edition-/java/com/example/androidfirstjavagame/Story.java
package com.example.androidfirstjavagame;

import android.content.Intent;
import android.view.View;

public class Story {
  
    GameScreen gs;
    String nextPosition1, nextPosition2, nextPosition3, nextPosition4;
    boolean gotSword=false;
    public Story(GameScreen gs){
        this.gs=gs;

    }


    public void selectPosition(String position){
        switch (position){
            case "startingPoint": startingPoint(); break;
            case "sign": sign(); break;
            case "monster":  monster();break;
            case "sword": sword(); break;
            case "pipe": pipe(); break;
            case "attack": attackMonster(); break;
            case "grave": death(); break;
            case "run": startingPoint(); break;
            case "reset": gs.resetGame(); break;
            case "lookInside": piranhaPlant(); break;



        }
    }

    public void startingPoint(){
        gs.image.setImageResource(R.drawable.trail);
        gs.text.setText("You are on the road.\nThere is a wooden sign nearby.\nWhat will You do?");
        gs.b1.setText("Go North");
        gs.b2.setText("Go East");
        gs.b3.setText("Go West");
        gs.b4.setText("Read sign");


        showAllButtons();

        nextPosition1="monster";
        nextPosition2="sword";
        nextPosition3="pipe";
        nextPosition4="sign";
    }




    public void sign(){
        gs.image.setImageResource(R.drawable.sign);
        gs.text.setText("The sign says: \n\nMONSTER AHEAD!");
        gs.b1.setText("Back");
        gs.b2.setText("");
        gs.b3.setText("");
        gs.b4.setText("");


        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.INVISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);


        nextPosition1="startingPoint";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";
    }

    public void monster(){

        gs.image.setImageResource(R.drawable.gargoyle);
        gs.text.setText("You encounter a gargoyle!!!");
        gs.b1.setText("Attack");
        gs.b2.setText("Run");
        gs.b3.setText("");
        gs.b4.setText("");


        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.VISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);


        nextPosition1="attack";
        nextPosition2="run";
        nextPosition3="";
        nextPosition4="";
    }

    public void attackMonster(){
        if(gotSword==false)
        {
        gs.image.setImageResource(R.drawable.gargoyle);
        gs.text.setText("You fight well but the monster\n is too strong for You! The monster kills You");
        gs.b1.setText(">");
        gs.b2.setText("");
        gs.b3.setText("");
        gs.b4.setText("");
        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.INVISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);
        nextPosition1="grave";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";
        }else{
            gs.image.setImageResource(R.drawable.chest);
            gs.text.setText("With your legendary swoard and experience as a warrior the monster has no chance to win!\nTHE END");
            gs.b1.setText("Start Again");
            gs.b2.setText("");
            gs.b3.setText("");
            gs.b4.setText("");
            gs.b1.setVisibility(View.VISIBLE);
            gs.b2.setVisibility(View.INVISIBLE);
            gs.b3.setVisibility(View.INVISIBLE);
            gs.b4.setVisibility(View.INVISIBLE);

            nextPosition1="reset";
            nextPosition2="";
            nextPosition3="";
            nextPosition4="";
            gotSword=false;

        }


    }

    public void death(){
        gs.image.setImageResource(R.drawable.grave);
        gs.text.setText("You are dead.\nYour adventure ends here.\nGAME OVER");
        gs.b1.setText("Start Again");
        gs.b2.setText("");
        gs.b3.setText("");
        gs.b4.setText("");
        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.INVISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);
        nextPosition1="reset";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";
        gotSword=false;
    }
    public void pipe(){
        gs.image.setImageResource(R.drawable.pipe);
        gs.text.setText("You find a gigantic pipe.");
        gs.b1.setText("Look inside");
        gs.b2.setText("Go back");
        gs.b3.setText("");
        gs.b4.setText("");


        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.VISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);


        nextPosition1="lookInside";
        nextPosition2="run";
        nextPosition3="";
        nextPosition4="";
    }

    public void piranhaPlant(){
        gs.image.setImageResource(R.drawable.plant);
        if(gotSword==false){

        gs.text.setText("Piranha plant is inside!!!\n You are eaten by it.");
        gs.b1.setText(">");
        gs.b2.setText("");
        gs.b3.setText("");
        gs.b4.setText("");
        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.INVISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);
        nextPosition1="grave";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";
        }else{
            gs.text.setText("You have defeated the Piranha plant\nwith your Master Sword!!!\nYou feel much stronger now");
            gs.b1.setText("Go Back");
            gs.b2.setText("");
            gs.b3.setText("");
            gs.b4.setText("");
            gs.b1.setVisibility(View.VISIBLE);
            gs.b2.setVisibility(View.INVISIBLE);
            gs.b3.setVisibility(View.INVISIBLE);
            gs.b4.setVisibility(View.INVISIBLE);
            nextPosition1="startingPoint";
            nextPosition2="";
            nextPosition3="";
            nextPosition4="";
        }
    }
    public void sword(){
        gs.image.setImageResource(R.drawable.sword);
        gs.text.setText("Amazing! You find a Master Sword! \nYou can kill Monster!");

        gs.b1.setText("Go Back");
        gs.b2.setText("");
        gs.b3.setText("");
        gs.b4.setText("");
        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.INVISIBLE);
        gs.b3.setVisibility(View.INVISIBLE);
        gs.b4.setVisibility(View.INVISIBLE);
        nextPosition1="startingPoint";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";

        gotSword=true;
    }
    public void showAllButtons(){
        gs.b1.setVisibility(View.VISIBLE);
        gs.b2.setVisibility(View.VISIBLE);
        gs.b3.setVisibility(View.VISIBLE);
        gs.b4.setVisibility(View.VISIBLE);
    }
}

File: Android Basic/Android-First-Kotlin-Game/java/com/example/myfirstandroidkotlingame/Story.kt
package com.example.myfirstandroidkotlingame

import android.view.View
import kotlinx.android.synthetic.main.activity_game_screen.*

class Story (val gs: GameScreen){
    // to wyzej to jak łaczymy obiekty
    //var - zmienna
    // val -  tworzymy dla czegos co sie nie zmienia
    // fun - function

    /*
        var number = 1;
    var text = "This is text without writing String before name"
     */
    var nextPosition1=""
    var nextPosition2=""
    var nextPosition3=""
    var nextPosition4=""

    var gotSword=false





     fun selectPosition(position: String?){
        when (position){
            "startingPoint"-> startingPoint();
             "sign"-> sign()
             "monster"->  monster()
             "sword"-> sword()
             "pipe"-> pipe()
             "attack"-> attackMonster()
             "grave"-> death()
             "run"-> startingPoint()
            "reset"->gs.resetGame()
             "lookInside"-> piranhaPlant()



        }
    }

     fun startingPoint(){
        gs.GameImageView.setImageResource(R.drawable.trail);
        gs.StoryTextView.setText("You are on the road.\nThere is a wooden sign nearby.\nWhat will You do?");
        gs.button1.setText("Go North");
        gs.button2.setText("Go East");
        gs.button3.setText("Go West");
        gs.button4.setText("Read sign");


        showAllButtons();

        nextPosition1="monster";
        nextPosition2="sword";
        nextPosition3="pipe";
        nextPosition4="sign";
    }




     fun sign(){


         gs.GameImageView.setImageResource(R.drawable.sign);
         gs.StoryTextView.setText("The sign says: \n\nMONSTER AHEAD!");
         gs.button1.setText("Back");
         gs.button2.setText("");
         gs.button3.setText("");
         gs.button4.setText("");


        gs.button1.setVisibility(View.VISIBLE);
        gs.button2.setVisibility(View.INVISIBLE);
        gs.button3.setVisibility(View.INVISIBLE);
        gs.button4.setVisibility(View.INVISIBLE);


        nextPosition1="startingPoint";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";
    }

     fun monster(){


         gs.GameImageView.setImageResource(R.drawable.gargoyle);
         gs.StoryTextView.setText("You encounter a gargoyle!!!");
         gs.button1.setText("Attack");
         gs.button2.setText("Run");
         gs.button3.setText("");
         gs.button4.setText("");


         gs.button1.setVisibility(View.VISIBLE);
         gs.button2.setVisibility(View.VISIBLE);
         gs.button3.setVisibility(View.INVISIBLE);
         gs.button4.setVisibility(View.INVISIBLE);




        nextPosition1="attack";
        nextPosition2="run";
        nextPosition3="";
        nextPosition4="";
    }

     fun attackMonster(){
        if(gotSword==false)
        {

            nextPosition1="grave";
            nextPosition2="";
            nextPosition3="";
            nextPosition4="";

            gs.GameImageView.setImageResource(R.drawable.gargoyle);
            gs.StoryTextView.setText("You fight well but the monster\n" + " is too strong for You! The monster kills You");
            gs.button1.setText(">");
            gs.button2.setText("");
            gs.button3.setText("");
            gs.button4.setText("");


            gs.button1.setVisibility(View.VISIBLE);
            gs.button2.setVisibility(View.INVISIBLE);
            gs.button3.setVisibility(View.INVISIBLE);
            gs.button4.setVisibility(View.INVISIBLE);



        }else{
            gs.GameImageView.setImageResource(R.drawable.chest);


            nextPosition1="reset";
            nextPosition2="";
            nextPosition3="";
            nextPosition4="";
            gotSword=false;




            gs.StoryTextView.setText("With your legendary swoard and experience as a warrior the monster has no chance to win!\n" + "THE END");
            gs.button1.setText("Start Again");
            gs.button2.setText("");
            gs.button3.setText("");
            gs.button4.setText("");


            gs.button1.setVisibility(View.VISIBLE);
            gs.button2.setVisibility(View.INVISIBLE);
            gs.button3.setVisibility(View.INVISIBLE);
            gs.button4.setVisibility(View.INVISIBLE);

        }


    }

     fun death(){
        gs.GameImageView.setImageResource(R.drawable.grave);
        gs.StoryTextView.setText("You are dead.\nYour adventure ends here.\nGAME OVER");
        gs.button1.setText("Start Again");
        gs.button2.setText("");
        gs.button3.setText("");
        gs.button4.setText("");
        gs.button1.setVisibility(View.VISIBLE);
        gs.button2.setVisibility(View.INVISIBLE);
        gs.button3.setVisibility(View.INVISIBLE);
        gs.button4.setVisibility(View.INVISIBLE);
        nextPosition1="reset";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";
        gotSword=false;


    }
     fun pipe(){
        gs.GameImageView.setImageResource(R.drawable.pipe);
        gs.StoryTextView.setText("You find a gigantic pipe.");
        gs.button1.setText("Look inside");
        gs.button2.setText("Go back");
        gs.button3.setText("");
        gs.button4.setText("");


        gs.button1.setVisibility(View.VISIBLE);
        gs.button2.setVisibility(View.VISIBLE);
        gs.button3.setVisibility(View.INVISIBLE);
        gs.button4.setVisibility(View.INVISIBLE);


        nextPosition1="lookInside";
        nextPosition2="run";
        nextPosition3="";
        nextPosition4="";
    }

     fun piranhaPlant(){
        gs.GameImageView.setImageResource(R.drawable.plant);
        if(gotSword==false){

            gs.StoryTextView.setText("Piranha plant is inside!!!\n You are eaten by it.");
            gs.button1.setText(">");
            gs.button2.setText("");
            gs.button3.setText("");
            gs.button4.setText("");
            gs.button1.setVisibility(View.VISIBLE);
            gs.button2.setVisibility(View.INVISIBLE);
            gs.button3.setVisibility(View.INVISIBLE);
            gs.button4.setVisibility(View.INVISIBLE);
            nextPosition1="grave";
            nextPosition2="";
            nextPosition3="";
            nextPosition4="";
        }else{
            gs.StoryTextView.setText("You have defeated the Piranha plant\nwith your Master Sword!!!\nYou feel much stronger now");
            gs.button1.setText("Go Back");
            gs.button2.setText("");
            gs.button3.setText("");
            gs.button4.setText("");
            gs.button1.setVisibility(View.VISIBLE);
            gs.button2.setVisibility(View.INVISIBLE);
            gs.button3.setVisibility(View.INVISIBLE);
            gs.button4.setVisibility(View.INVISIBLE);
            nextPosition1="startingPoint";
            nextPosition2="";
            nextPosition3="";
            nextPosition4="";
        }
    }
     fun sword(){
        gs.GameImageView.setImageResource(R.drawable.sword);
        gs.StoryTextView.setText("Amazing! You find a Master Sword! \nYou can kill Monster!");

        gs.button1.setText("Go Back");
        gs.button2.setText("");
        gs.button3.setText("");
        gs.button4.setText("");
        gs.button1.setVisibility(View.VISIBLE);
        gs.button2.setVisibility(View.INVISIBLE);
        gs.button3.setVisibility(View.INVISIBLE);
        gs.button4.setVisibility(View.INVISIBLE);
        nextPosition1="startingPoint";
        nextPosition2="";
        nextPosition3="";
        nextPosition4="";

        gotSword=true;
    }
     fun showAllButtons(){
        gs.button1.setVisibility(View.VISIBLE);
        gs.button2.setVisibility(View.VISIBLE);
        gs.button3.setVisibility(View.VISIBLE);
        gs.button4.setVisibility(View.VISIBLE);
    }
}


